_posts/en/newsletters/2023-10-25-newsletter.md

```diff
@@ -0,0 +1,369 @@
+  promptly, Bob would learn the preimage and be able to claim the
+  forwarded payment from MalloryA, making Bob happy.
+
+  However, if MalloryB replaces Bob's spend and a spend of her own that
+  contains the preimage but is also a removable tranaction, and she
+  quickly removes that transaction, then it's unlikely that either Bob's
+  spend or MalloryB's preimage will appear in the block chain.  This
+  prevents Bob from getting his money back from MalloryB.  Without the
+  preimage, the trustless LN protocol prevents Bob from being able to keep
+  the forwarded payment from MalloryA, so he gives her a refund.  At this
+  point, MalloryB gets her spend containing the preimage confirmed,
+  allowing her to claim the forwarded payment from Bob.  That means, if an
+  amount of _x_ was forwarded, MalloryA pays zero, MalloryB receives _x_,
+  and Bob loses _x_ (not counting various fees).
+
+  For the attack to be profitable, MalloryB must share a channel with
+  Bob---but MalloryA can be anywhere along the forwarding path to Bob.
+  For example:
+
+    MalloryA -> X -> Y -> Z -> Bob -> MalloryB
+
+  Replacement cycling has similar consequences for LN nodes to existing
+  [transaction pinning attacks][].  However, techniques such as [v3
+  transaction relay][] that were designed to prevent pinning for LN and
+  similar protocols do not prevent replacement cycling.
+
+- **Deployed mitigations in LN nodes for replacement cycling:** as
+  [described][] by FIXME:ariad, several mitigations have been deployed
+  by LN implementations.
+
+    - **Frequent rebroadcasting:** after a relay node's mempool has
+      Bob's spend replaced by Mallory's spend, and then has Mallory's
+      spend removed by Mallory's replacement of the precursor
+      transaction, that relay node will immediately be willing to accept
+      Bob's spend again.  All Bob needs to do is re-broadcast his spend,
+      which costs him nothing beyond the transaction fee he was already
+      willing to pay.
+
+      Before the private disclosure of replacement cycling, LN
+      implementations only rebroadcasted their transactions infrequently
+      (once per block or less).  There's normally a [privacy cost][topic
+      transaction origin privacy] to broadcasting and rebroadcasting
+      transactions---it might make it easier for third parties to
+      associate Bob's LN activities with his IP address---although few
+      public LN forwarding nodes currently try to hide this.  Now Core
+      Lightning, Eclair, LDK, and LND will all rebroadcast more
+      frequently.
+
+      After each time Bob rebroadcasts, Mallory can use the same
+      technique to replace his transaction again.  However, the BIP125
+      replacement rules will require Mallory to pay additional
+      transaction fees for each of her replacements and removals,
+      meaning each rebroadcast by Bob lowers the profitability to
+      Mallory of a successful attack.
```

Comment:
Since Bob’s transaction always has the same feerate and that feerate must be uncompetitive for the attack to be viable, is it really necessary for Mallory to increasingly pay more fees? If e.g. Mallory operates a high-volume wallet, she could use a UTXO to do the replacement cycle and then use that input towards an actual transaction at a competitive feerate.

Reply:
> Since Bob’s transaction always has the same feerate and that feerate must be uncompetitive for the attack to be viable

By "uncompetitive", do you mean that Bob must be paying too low of a feerate?  That's not actually true.  Bob could pay a feerate of 1,000 sat/vb and Mallory would still be able to perform a replacement cycle, which might still be worth it if earns her a 1,000,000 sat HTLC.

Was it something I wrote that gave you the impression that Bob's feerate must be uncompetitive?

> If e.g. Mallory operates a high-volume wallet, she could use a UTXO to do the replacement cycle and then use that input towards an actual transaction at a competitive feerate.

That's correct, but it's also where the deployed mitigations and proposed mitigations come in.  If Bob rebroadcasts 5 times a block, Mallory will need to have an operation where shes send about 5 organic transactions per block in order to have a marginal cost of zero.  That's fairly high volume, and it scales almost linearly depending on how many HTLCs Mallory is attacking (e.g., if she had 10 HTLCs in the channel with Bob, she would need to be at the level of typically sending about 50 organic transactions per block).  The additional proposed mitigation of having Bob pay higher fees as the CLTV expiry delta eliminates this advantage.  If Bob pays 10x the next-block fee near the end of CLTV expiry delta, then Mallory's replacements will need to pay an absolute fee greater than that amount, which she can only spread among additional outputs in organic transactions up to the 100,000 vbyte limit.  After that, she'll need to overpay fees.

In other words, I *think* the mitigations quickly make it so that only the highest of high-frequency spenders could have a minimal marginal attack cost, and that would only be possible if they only attacked a few HTLCs at once.  If they attacked more than about 100 HTLCs simultaneously (which is just 1 HTLC per block with a CLTV expiry delta of 144), I think anyone would be guaranteed to pay an above-zero marginal cost.